shell/platform/android/apk_asset_provider.cc

```diff
@@ -0,0 +1,38 @@
+#include <unistd.h>
+#include <algorithm>
+#include <sstream>
+
+#include "flutter/shell/platform/android/apk_asset_provider.h"
+#include "lib/fxl/logging.h"
+
+namespace blink {
+
+bool APKAssetProvider::GetAsBuffer(const std::string& asset_name,
+                                   std::vector<uint8_t>* data) {
+  std::stringstream ss;
+  ss << directory_.c_str() << "/" << asset_name;
+  AAsset* asset = AAssetManager_open(assetManager_, ss.str().c_str(), AASSET_MODE_BUFFER);
+  if (!asset) {
+     return false;
+  }
+
+  uint8_t* buffer = (uint8_t*)AAsset_getBuffer(asset);
+  if (!buffer) {
+    FXL_LOG(ERROR) << "Got null trying to acquire buffer for asset:" << asset;
+    return false;
```

Comment:
Missing `AAset_close` in case of this error. The idiom is to use `fxl::UniqueObject` for such calls but I can certainly understand if you don't want to write that for this one-of case and just add the close call manually here.